index.js

```diff
@@ -180,6 +181,97 @@ app.get('/make-guess/:user_id/:guess', async (req, res) => {
 });
 
 
+app.get('/make-guess/v2/:user_id/:guess', async (req, res) => {
+  userID = req.params.user_id;
+  guess = req.params.guess.toLowerCase(); // lowercase letters only
+  if (guess.length !== 1) {
+    // should be a single letter
+    res.status(400).send([]) // 400 bad request
+    return
+  }
+  if (!alphabetLower.includes(guess)) {
+    // guess not a letter
+    res.status(400).send([]) // 400 bad request
+    return
+  }
+  const data_file = fs.readFileSync('./leaderboard.json', 'utf-8');
+  const data_json = JSON.parse(data_file);
+  
+  var entry = data_json.find(entry => entry.user_id == userID);
+  var index = data_json.indexOf(entry)
+  console.log(index)
+  if (entry.guesses_left <= 0) {
+    // no more guesses left
+    res.status(200).json({"winner": false});
+    return
+  }
+
+  var word = entry.current_word
+
+  var guessedLetters = entry.guessing_word
+  console.log(guessedLetters)
+  if (guessedLetters.includes(guess)) {
+    console.log("already guessed")
+    res.status(400).send([]) // 400 bad request
+    return
+  }
+
+  // guessing_word is the correct guesses
+  if (word.includes(guess)) {
+    // Word contains the guess, do not dock guess and renew 'guessing_word'
+    var guessingWord = data_json[index].guessing_word
+
+    // TODO: need guessed_letters in the database
+    // This will make it easy also to keep track of already guessed letters e.g. both correct/incorrect guesses to be disabled on the frontend
```

Comment:
I think we need to keep track of `guessed_letters` in the database as well. I noticed yours wasn't complete but maybe you worked on it? If not I'll continue on V2 tomorrow and add this as a field